File: android/app/src/main/kotlin/com/agent/aios/agent/tools/ExtendedTool.kt
package com.agent.aios.agent.tools

import com.agent.aios.domain.ToolContext

interface ExtendedTool {
    val name: String
    val description: String
    val parameters: String

    fun execute(
        args: String,
        toolContext: ToolContext,
    ): String
}

File: lib/domain/agent/extended_tool.dart
import 'package:aios/domain/agent/tool_context.dart';
import 'package:aios/domain/agent/tool_result.dart';

abstract class ExtendedTool {
  String get name;
  String get description;
  String get parameters;

  String get toolPrompt => '$description\nParameters: $parameters';

  Future<ToolResult> execute(String args, ToolContext toolContext);

  Future<String?> validate(String args, ToolContext toolContext) async => null;
}
